journal/zap.go

```diff
@@ -0,0 +1,51 @@
+package journal
+
+import (
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// NewZapJournal returns a Builder backed by a zap logger.
+func NewZapJournal(filepath string) (Journal, error) {
+	zapCfg := zap.NewProductionConfig()
+	zapCfg.Encoding = "json"
+	zapCfg.EncoderConfig.EncodeTime = zapcore.ISO8601TimeEncoder
+	zapCfg.EncoderConfig.LevelKey = ""
+	zapCfg.EncoderConfig.CallerKey = ""
+	zapCfg.EncoderConfig.MessageKey = "operation"
+	zapCfg.EncoderConfig.NameKey = "topic"
```

Comment:
To avoid name collisions, I recommend a prefix, e.g. `_topic`. And "event" rather than "operation"?